app/code/Magento/AwsS3/Model/Cached/CachedAdapter.php

```diff
@@ -0,0 +1,308 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AwsS3\Model\Cached;
+
+use League\Flysystem\Config;
+use League\Flysystem\FileAttributes;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\FilesystemException;
+use Magento\RemoteStorage\Driver\Cache\CacheInterface;
+use Magento\RemoteStorage\Model\GetPathInfo;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Aws cached adapter model.
+ */
+class CachedAdapter implements FilesystemAdapter
+{
+    /**
+     * @var FilesystemAdapter
+     */
+    private FilesystemAdapter $adapter;
```

Comment:
We should still support PHP 7.3. I don't think it supports parameter types

Reply:
Removed.